packages/ionic/src/layouts/categorization/categorization-tab-layout.ts

```diff
@@ -29,44 +30,68 @@ import { CategoryRenderer } from './category/category';
 })
 export class CategorizationTabLayoutRenderer extends JsonFormsIonicLayout {
 
-    categoryPages: any[];
+  categoryPages: any[];
 
-    constructor(ngRedux: NgRedux<JsonFormsState>) {
-        super(ngRedux);
-        this.categoryPages = [];
-        this.initializers.push(this.mapAdditionalProps);
-    }
+  constructor(ngRedux: NgRedux<JsonFormsState>, private paramsService: ParamsService) {
+    super(ngRedux);
+    this.categoryPages = [];
+    this.initializers.push(this.mapAdditionalProps);
+  }
 
-    mapAdditionalProps = (props: any) => {
-        this.categoryPages = [];
-        (props.uischema as Categorization).elements.forEach(
-            (category, index) => {
-                if (this.categoryPages[index] === undefined ||
-                    !_.isEqual(this.categoryPages[index].params.category, category)) {
-                    this.categoryPages.push({
-                        renderer: CategoryRenderer,
-                        params: {
-                            category: {
-                                elements: category.elements,
-                                label: category.label,
-                                schema: this.schema,
-                                path: this.path
-                            }
-                        }
-                    });
-                }
+  mapAdditionalProps = (props: any) => {
+    if (this.categoryPages.length === 0) {
```

Comment:
I don't get why we need to do so complicated stuff and can't simply rely on angular/ionic and the trackBy function to render the right stuff?